lang/syn/src/codegen/error.rs

```diff
@@ -41,49 +56,29 @@ pub fn generate(error: Error) -> proc_macro2::TokenStream {
     };
 
     quote! {
-        /// Anchor generated Result to be used as the return type for the
-        /// program.
-        pub type Result<T> = std::result::Result<T, Error>;
-
-        /// Anchor generated error allowing one to easily return a
-        /// `ProgramError` or a custom, user defined error code by utilizing
-        /// its `From` implementation.
-        #[doc(hidden)]
-        #[derive(thiserror::Error, Debug)]
-        pub enum Error {
-            #[error(transparent)]
-            ProgramError(#[from] anchor_lang::solana_program::program_error::ProgramError),
-            #[error(transparent)]
-            ErrorCode(#[from] #enum_name),
-        }
-
         #[derive(std::fmt::Debug, Clone, Copy)]
         #[repr(u32)]
         #error_enum
 
-        impl std::fmt::Display for #enum_name {
-            fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        impl #enum_name {
+            pub fn name(&self) -> String {
                 match self {
-                    #(#variant_dispatch),*
+                    #(#name_variant_dispatch),*
                 }
             }
         }
 
-        impl std::error::Error for #enum_name {}
-
-        impl std::convert::From<Error> for anchor_lang::solana_program::program_error::ProgramError {
-            fn from(e: Error) -> anchor_lang::solana_program::program_error::ProgramError {
-                match e {
-                    Error::ProgramError(e) => e,
-                    Error::ErrorCode(c) => anchor_lang::solana_program::program_error::ProgramError::Custom(c as u32 + #offset),
-                }
+        impl From<#enum_name> for u32 {
+            fn from(e: #enum_name) -> u32 {
+                e as u32 + #offset
             }
         }
 
-        impl std::convert::From<#enum_name> for anchor_lang::solana_program::program_error::ProgramError {
```

Comment:
Can we simplify a lot of changes here if we `impl From<#enum_name>  for `anchor_lang::Error` where we use `anchor_lang::error_without_origin` in the implementation?

Reply:
For example changes like line 81 in `lang/syn/src/codegen/program/cpi.rs` can be undone.

Reply:
yea I think that would work Ill try